src/lingrex/trimming.py

```diff
@@ -0,0 +1,201 @@
+        exclude="_+"
+        ):
+    """
+    Trim alignment sites by gaps, preserving a core of sites.
+
+    :parameter alms: Alignment sites of a cognate set.
+    :type alms: list
+    :parameter threshold: Threshold by which sites with gaps should be trimmed.
+        Defaults to '0.5'.
+    :type threshold: int
+    :param skeletons: Tuple of syllable-skeletons that should be preserved
+        for further processing. Defaults to '("CV", "VC")'.
+    :type skeletons: tuple
+    :parameter gap: String that codes gaps in alignment sites. Defaults to '-'.
+    :type gap: string
+    :return: Indices of trimmed strings.
+    :rtype: set
+    """
+    cons = [gap if c >= threshold else "S" for c in gap_profile(alms, gap=gap)]
+    skeleton = get_skeleton(alms, gap=gap)
+    left, right = consecutive_gaps(cons)
+
+    idxs = [i for i, c in enumerate(skeleton) if c in exclude]
+
+    # order by first
+    sorted_indices = right[::-1] + left
+    while sorted_indices:
+        idx = sorted_indices.pop(0)
+        current_skeleton = "".join([skeleton[i] for i in
+                                    range(len(skeleton)) if i not in idxs+[idx]])
+        if any([subsequence_of(s, current_skeleton) for s in skeletons]):
+            idxs += [idx]
+        else:
+            break
+    return sorted(set(idxs))
+
+
+def trim_random(
+        alms,
+        func=None,
+        threshold=0.5,
+        skeletons=("CV", "VC"),
+        gap="-",
+        exclude="_+",
+        ):
+    """
+    For a base trim function, return a random version with a similar CV distribution.
+
+    :parameter alms: Alignment sites of a cognate set.
+    :type alms: list
+    :parameter func: Trimming function that should be applied. Defaults to 'None'.
+    :type func: function
+    :parameter threshold: Threshold by which sites with gaps should be trimmed.
+        Defaults to '0.5'.
+    :type threshold: int
+    :param skeletons: Tuple of syllable-skeletons that should be preserved
+        for further processing. Defaults to '("CV", "VC")'.
+    :type skeletons: tuple
+    :parameter gap: String that codes gaps in alignment sites. Defaults to '-'.
+    :type gap: string
+    :return: Indices of trimmed strings.
+    :rtype: set
+    """
+    func = func or trim_by_gap
+    reference = apply_trim(
+            alms,
+            func(
+                alms, threshold=threshold, skeletons=skeletons, gap=gap,
+                exclude=exclude)
+            )
+    reference_skeleton = get_skeleton(reference)
+    # create a freq dict of ref skel
+    rs_freqs = collections.defaultdict(int)
+    for c in reference_skeleton:
+        rs_freqs[c] += 1
+    # get a dictionary of indices by position
+    indices = collections.defaultdict(list)
+    for i, c in enumerate(get_skeleton(alms)):
+        indices[c] += [i]
+    # random sample indices to be retained
+    retain = []
+    for c, _ in rs_freqs.items():
+        retain += random.sample(indices[c], rs_freqs[c])
```

Comment:
I'd use list comprehension to construct `retain`:
```python
retain = [random.sample(indices[c], rs_freqs[c]) for c, _ in rs_freqs.items()]
```
Looks more transparent to me.
(I also don't like `+=` instead of `list.append()` in such cases - but I that might just be a presonal preference.)